sqlglot/optimizer/eliminate_subqueries.py

```diff
@@ -1,11 +1,20 @@
+from __future__ import annotations
+
 import itertools
+import typing as t
 
 from sqlglot import expressions as exp
 from sqlglot.helper import find_new_name
-from sqlglot.optimizer.scope import build_scope
+from sqlglot.optimizer.scope import Scope, build_scope
+
+if t.TYPE_CHECKING:
+    from sqlglot._typing import E
+
+    ExistingCTEsMapping = t.Dict[E, str]
+    TakenNameMaping = t.Dict[t.Any, t.Union[Scope, E]]
```

Comment:
There's a typo here, it needs to be "Mapping". Additionally, why are we using `t.Any` for the key type?

Reply:
Hm, was sure that it was complaining about a `t.Any` entry somewhere but I changed it back to `str` and all seems good